pkg/controller/nodes/task/resourcemanager/redis_resourcemanager.go

```diff
@@ -203,19 +230,36 @@ func (r *RedisResourceManager) AllocateResource(ctx context.Context, namespace p
 		return pluginCore.AllocationStatusGranted, nil
 	}
 
-	size, err := r.client.SCard(string(namespace)).Result()
+	size, err := r.client.SCard(string(namespace))
 	if err != nil {
 		logger.Errorf(ctx, "Error getting size of Redis set %v", err)
 		return pluginCore.AllocationUndefined, err
 	}
 
-	if size > int64(namespacedResource.quota) {
+	if size >= int64(namespacedResource.quota) {
```

Comment:
nit: should this be represented as a "constraint" as well and use the globalConstraint.IsAllowed(size) check for consistency?

Reply:
refactored